[Android] Version 2.10.8-2.12.3 crashes on Configuration button click. Partially fixed in version 2.13.10.

When I click the Configuration button, Joplin app crashes with StackOverflow exception. The issue occurred both on the full notes database from the prior Joplin version (after the app update) and on the clean database (after deleting the app data).

## Environment

Joplin version: 2.10.8-2.12.3
Platform: Android
OS specifics: Lenovo TB-8704F tablet, Android 8.1.0 (SDK 27)
Partially fixed in version 2.13.10 (only some Configuration sections are crashing now).

## Steps to reproduce

1. Launch Joplin
2. Click Configuration button
3. Select one of following sections: Appearance, Synchronization, Note History.
4. The app crashes

## Describe what you expected to happen

The Configuration section opens.

## Logfile (probably obsolete in the current version)

Unfortunately, I cannot copy full stack trace from the Android app feedback preview popup. Here are some details:

Exception: java.lang.StackOverflowError
Source class: android.content.ContextWrapper
Source method: getApplicationInfo
Line number: 152
Stack trace excerpts:
com.facebook.react.bridge.JSApplicationIllegalArgumentException: error while updating property 'step' of a view managed by RNCSlider
at com.facebook.react.urlmanager.ViewManagersPropertyCache$PropSetter.updateViewProp(ViewManagerPropertyCache.java:101)
...
Caused by: java.lang.StackOverflowError: stack size 8MB
at android.content.ContextWrapper.getApplicationInfo(ContextWrapper.java:152)
...
at android.widget.ProgressBar.setProgress(ProgressBar.java:1385)
at com.reactnativecommunity.slider.ReactSliderManager$1.onProgressChanged(ReactSliderManager.java:38)
at android.widget.ProgressBar.onProgressRefresh(SeekBar.java:95)
at android.widget.ProgressBar.doRefreshProgress(ProgressBar.java:11298)
at android.widget.ProgressBar.refreshProgress(ProgressBar.java:1353)
at android.widget.ProgressBar.setProgressInternal(ProgressBar.java:1418)
at android.widget.ProgressBar.setProgress(ProgressBar.java:1385)

(The last part loops until stack overflow).

--- a/.yarn/patches/@react-native-community-slider-npm-4.4.4-d78e472f48.patch
+++ b/.yarn/patches/@react-native-community-slider-npm-4.4.4-d78e472f48.patch
@@ -1,0 +1,62 @@
+diff --git a/android/src/newarch/java/com/reactnativecommunity/slider/ReactSliderManager.java b/android/src/newarch/java/com/reactnativecommunity/slider/ReactSliderManager.java
+index a5bb95eec3337b93a2338a2869a2bda176c91cae..87817688eb280c2f702c26dc35558c6a0a4db1ea 100644
+--- a/android/src/newarch/java/com/reactnativecommunity/slider/ReactSliderManager.java
++++ b/android/src/newarch/java/com/reactnativecommunity/slider/ReactSliderManager.java
+@@ -42,12 +42,20 @@ public class ReactSliderManager extends SimpleViewManager<ReactSlider> implement
+             public void onProgressChanged(SeekBar seekbar, int progress, boolean fromUser) {
+               ReactSlider slider = (ReactSlider)seekbar;
+ 
+-              if(progress < slider.getLowerLimit()) {
+-                progress = slider.getLowerLimit();
+-                seekbar.setProgress(progress);
+-              } else if (progress > slider.getUpperLimit()) {
+-                progress = slider.getUpperLimit();
+-                seekbar.setProgress(progress);
++              // During initialization, lowerLimit can be greater than upperLimit.
++              //
++              // If a change event is received during this, we need a check to prevent
++              // infinite recursion.
++              //
++              // Issue: https://github.com/callstack/react-native-slider/issues/571
++              if (slider.getLowerLimit() <= slider.getUpperLimit()) {
++                if(progress < slider.getLowerLimit()) {
++                  progress = slider.getLowerLimit();
++                  seekbar.setProgress(progress);
++                } else if (progress > slider.getUpperLimit()) {
++                  progress = slider.getUpperLimit();
++                  seekbar.setProgress(progress);
++                }
+               }
+ 
+               ReactContext reactContext = (ReactContext) seekbar.getContext();
+diff --git a/android/src/oldarch/java/com/reactnativecommunity/slider/ReactSliderManager.java b/android/src/oldarch/java/com/reactnativecommunity/slider/ReactSliderManager.java
+index 3ff5930f85a3cd92c2549925f41058abb188a57e..ab3681fdfe0b736c97020e1434e450c8183e6f18 100644
+--- a/android/src/oldarch/java/com/reactnativecommunity/slider/ReactSliderManager.java
++++ b/android/src/oldarch/java/com/reactnativecommunity/slider/ReactSliderManager.java
+@@ -30,12 +30,20 @@ public class ReactSliderManager extends SimpleViewManager<ReactSlider> {
+             public void onProgressChanged(SeekBar seekbar, int progress, boolean fromUser) {
+               ReactSlider slider = (ReactSlider)seekbar;
+ 
+-              if(progress < slider.getLowerLimit()) {
+-                progress = slider.getLowerLimit();
+-                seekbar.setProgress(progress);
+-              } else if(progress > slider.getUpperLimit()) {
+-                progress = slider.getUpperLimit();
+-                seekbar.setProgress(progress);
++              // During initialization, lowerLimit can be greater than upperLimit.
++              //
++              // If a change event is received during this, we need a check to prevent
++              // infinite recursion.
++              //
++              // Issue: https://github.com/callstack/react-native-slider/issues/571
++              if (slider.getLowerLimit() <= slider.getUpperLimit()) {
++                if(progress < slider.getLowerLimit()) {
++                  progress = slider.getLowerLimit();
++                  seekbar.setProgress(progress);
++                } else if (progress > slider.getUpperLimit()) {
++                  progress = slider.getUpperLimit();
++                  seekbar.setProgress(progress);
++                }
+               }
+ 
+               ReactContext reactContext = (ReactContext) seekbar.getContext();

--- a/package.json
+++ b/package.json
@@ -107,6 +107,7 @@
     "app-builder-lib@24.4.0": "patch:app-builder-lib@npm%3A24.4.0#./.yarn/patches/app-builder-lib-npm-24.4.0-05322ff057.patch",
     "react-native@0.71.10": "patch:react-native@npm%3A0.71.10#./.yarn/patches/react-native-animation-fix/react-native-npm-0.71.10-f9c32562d8.patch",
     "nanoid": "patch:nanoid@npm%3A3.3.7#./.yarn/patches/nanoid-npm-3.3.7-98824ba130.patch",
-    "pdfjs-dist": "patch:pdfjs-dist@npm%3A3.11.174#./.yarn/patches/pdfjs-dist-npm-3.11.174-67f2fee6d6.patch"
+    "pdfjs-dist": "patch:pdfjs-dist@npm%3A3.11.174#./.yarn/patches/pdfjs-dist-npm-3.11.174-67f2fee6d6.patch",
+    "@react-native-community/slider@4.4.4": "patch:@react-native-community/slider@npm%3A4.4.4#./.yarn/patches/@react-native-community-slider-npm-4.4.4-d78e472f48.patch"
   }
 }

